mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,175 @@
+	var newReviewer string
+
+	if issue.Assignees != nil && len(issue.Assignees) > 0 {
+		for _, oldReviewer := range issue.Assignees {
+			login := *oldReviewer.Login
+
+			for potentialOwner := range potentialOwners {
+				if login == potentialOwner {
+					weightSum -= potentialOwners[login]
+					delete(potentialOwners, login)
+					break
+				}
+			}
+		}
+	}
+
+	if len(potentialOwners) == 0 {
+		return newReviewer
+	}
+
+	newReviewer = selectMultipleOwners(potentialOwners, weightSum, 1)[0]
+
+	return newReviewer
+}
+
+// Munge is the workhorse encouraging PR author to assign a new reviewer
+// after getting no response from current reviewer for one week
+// The algorithm:
+// (1) find last modification time of the reviewer
+// (2) if the date is one week or longer before today's date, create a comment
+//     encouraging the author to assign a new reviewer and unassign the old reviewer
+// (3) suggest the new reviewer using Blunderbuss algorithm, making sure the old reviewer is not suggested
+// Note: the munger will post at most 5 comments
+func (i *InactiveReviewHandler) Munge(obj *github.MungeObject) {
+	issue := obj.Issue
+
+	// do not suggest new reviewer if it is not a PR or the PR has no author information
+	if !obj.IsPR() || issue.User == nil || issue.User.Login == nil {
+		return
+	}
+
+	comments, ok := obj.ListComments()
+	if !ok {
+		return
+	}
+
+	reviewComments, ok := obj.ListReviewComments()
+	if !ok {
+		return
+	}
+
+	if i.isPRActive(obj.Issue.CreatedAt, comments, reviewComments) {
+		return
+	}
+
+	files, ok := obj.ListFiles()
+	if !ok || len(files) == 0 {
+		glog.Errorf("failed to detect any changed file when assigning a new reviewer for inactive PR #%v", *obj.Issue.Number)
+		return
+	}
+
+	pinger := matchers.NewPinger("INACTIVE PR").SetTimePeriod(7 * 24 * time.Hour).SetMaxCount(5)
+	lastHumanModificationDate := i.getLastDate(obj.Issue.CreatedAt, comments, reviewComments, true)
+
+	if pinger.IsMaxReached(comments, lastHumanModificationDate) {
+		return
+	}
+
+	// only run Blunderbuss algorithm when ping limit is not reached
+	potentialOwners, weightSum := getAllPotentialOwners(issue.User.Login, i.features, files, issue.Number)
+	newReviewer := i.suggestNewReviewer(issue, potentialOwners, weightSum)
+	var msg string
+
+	if len(issue.Assignees) == 0 {
+		msg = fmt.Sprintf("To expedite a review, consider assigning _%s_.", newReviewer)
+	} else if len(newReviewer) == 0 {
+		msg = fmt.Sprintf("Sorry the review process for your PR has stalled. Your reviewer(s) may be on vacation or otherwise occupied. Consider pinging them.")
+	} else {
+		msg = fmt.Sprintf("Sorry the review process for your PR has stalled. Your reviewer(s) may be on vacation or otherwise occupied. Consider unassigning them using `/unassign` command, and assigning _%s_.", newReviewer)
+	}
+
+	notification := pinger.PingNotification(comments, msg, lastHumanModificationDate)
```

Comment:
yes. i removed it before due to `pinger.isMaxReached` check before. it is still possible to return null due to `shouldPingNow` call.